tests/samplers_tests/tpe_tests/test_sampler.py

```diff
@@ -783,54 +798,80 @@ def objective(trial: Trial) -> float:
         (-3, [float("inf")]),  # PRUNED (with a NaN intermediate value; it's treated as infinity)
         (float("inf"), [sign * 0.0]),  # PRUNED (without intermediate values)
     ]
-    assert _tpe.sampler._get_observation_pairs(study, ["x"], False) == (
+    violations = [0, 0, 1, 2] if constraints_enabled else None
+    assert _tpe.sampler._get_observation_pairs(
+        study, ["x"], False, constraints_enabled=constraints_enabled
+    ) == (
         {"x": [5.0, 5.0, 5.0, 5.0]},
         scores,
+        violations,
     )
-    assert _tpe.sampler._get_observation_pairs(study, ["y"], False) == (
+    assert _tpe.sampler._get_observation_pairs(
+        study, ["y"], False, constraints_enabled=constraints_enabled
+    ) == (
         {"y": [None, None, None, None]},
         scores,
+        violations,
     )
-    assert _tpe.sampler._get_observation_pairs(study, ["z"], False) == (
+    assert _tpe.sampler._get_observation_pairs(
+        study, ["z"], False, constraints_enabled=constraints_enabled
+    ) == (
         {"z": [0, 0, 0, 0]},  # The internal representation of 'None' for z is 0
         scores,
+        violations,
     )
-    assert _tpe.sampler._get_observation_pairs(study, ["x"], True) == (
+    assert _tpe.sampler._get_observation_pairs(
+        study, ["x"], True, constraints_enabled=constraints_enabled
+    ) == (
         {"x": [5.0, 5.0, 5.0, 5.0]},
         scores,
+        violations,
     )
-    assert _tpe.sampler._get_observation_pairs(study, ["y"], True) == ({"y": []}, [])
-    assert _tpe.sampler._get_observation_pairs(study, ["z"], True) == (
+    assert _tpe.sampler._get_observation_pairs(
+        study, ["y"], True, constraints_enabled=constraints_enabled
+    ) == ({"y": []}, [], [] if constraints_enabled else None)
```

Comment:
ditto. This test function also changes the expected values depending on the parametrized value.